File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#11 정렬/2108_통계학 (22.04.15 성공).py
from collections import Counter

N = int(input())
arr = []

for _ in range(N):
    num = int(input())
    arr.append(num)

arr.sort()

cnt = Counter(arr).most_common(2)

print(round(sum(arr) / N)) # 산술평균
print(arr[(N - 1) / 2]) # 중앙값
if N > 1: # 최빈값
    if cnt[0][1] == cnt[1][1]:
        print(cnt[1][0])
    else:
        print(cnt[0][0])
else:
    print(cnt[0][0])
print(max(arr) - min(arr)) # 최대 최소 차이

File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#11 정렬/2108_통계학 (22.04.15 성공).cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    int N, sum = 0, cntmax = 0, val = 0;
    cin >> N;
    vector<int> v(N); // 값을 저장할 벡터
    vector<int> cnt(8001); // 최빈값을 계산할 벡터

    for (int i = 0; i < N; ++i)  {
        cin >> v[i];
        sum += v[i]; // 총합 계산

        cnt[v[i] + 4000]++; // 최빈값 저장
    }

    sort(v.begin(), v.end());

    for (int i = 0; i <= 8000; ++i) {
        if (cnt[i] > cntmax) {
            cntmax = cnt[i];
            val = i;
        }
    }

    for (int i = val + 1; i <= 8000; ++i) {
        if (cnt[i] == cntmax) {
            val = i;
            break;
        }
    }
    sum = round(double(sum) / N);

    cout << sum << "\n"; // 산술평균, round(double(sum) / N)을 그대로 쓰면 -0으로 출력됨
    cout << v[(N - 1) / 2] << "\n"; // 중앙값
    cout << val - 4000 << "\n"; // 최빈값
    cout << v[N - 1] - v[0]; // 최대 최소 차이
}